File: test/spec/r/r_scripts/boolean.r
(should (Boolean true),
        (eq true))
(should (Boolean false),
        (eq false))
# equality
(should (equals? true, true),
        (eq true))
(should (equals? true, false),
        (eq false))
# complement
(should (complement true),
        (eq false))
(should (complement false),
        (eq true))
# conjunction
(should (conjunction true, true),
        (eq true))
(should (conjunction true, false),
        (eq false))
# disjunction
(should (disjunction true, true),
        (eq true))
(should (disjunction true, false),
        (eq true))
(should (disjunction false, false),
        (eq false))


        

File: lib/sbyc/r/system/boolean.rb
SByC::R::Boolean::Operators.define{

  operator {|op|
    op.description = %Q{ Computes boolean complement }
    op.signature   = [SByC::R::Boolean]
    op.argnames    = [:operand]
    op.returns     = SByC::R::Boolean
    op.aliases     = [:complement, :'!', :'~', :not]
  }
  def complement(operand); !operand end
  
  operator {|op|
    op.description = %Q{ Computes boolean conjunction }
    op.signature   = [SByC::R::Boolean, SByC::R::Boolean]
    op.argnames    = [:left, :right]
    op.returns     = SByC::R::Boolean
    op.aliases     = [:conjunction, :'&', :and]
  }
  def conjunction(left, right) (left & right) end
  
  operator {|op|
    op.description = %Q{ Computes boolean disjunction }
    op.signature   = [SByC::R::Boolean, SByC::R::Boolean]
    op.argnames    = [:left, :right]
    op.returns     = SByC::R::Boolean
    op.aliases     = [:disjunction, :'|', :or]
  }
  def disjunction(left, right) (left | right) end
  
}